src/features/defi/providers/osmosis/components/OsmosisManager/Withdraw/components/Withdraw.tsx

```diff
@@ -0,0 +1,534 @@
+      const tokenOutMins = await calculateTokenOutMins(
+        bnOrZero(formValues.cryptoAmount)
+          .multipliedBy(bn(10).pow(lpAsset?.precision ?? '0'))
+          .toString(),
+      )
+      if (!tokenOutMins) return
+      contextDispatch({
+        type: OsmosisWithdrawActionType.SET_WITHDRAW,
+        payload: {
+          underlyingAsset0: tokenOutMins[0],
+          underlyingAsset1: tokenOutMins[1],
+          shareInAmount: bnOrZero(formValues.cryptoAmount)
+            .multipliedBy(bn(10).pow(lpAsset?.precision ?? '0'))
+            .toString(),
+        },
+      })
+
+      const estimatedFeeCrypto = await getWithdrawFeeEstimate()
+      if (!estimatedFeeCrypto) {
+        contextDispatch({ type: OsmosisWithdrawActionType.SET_LOADING, payload: false })
+        return
+      }
+      contextDispatch({
+        type: OsmosisWithdrawActionType.SET_WITHDRAW,
+        payload: { estimatedFeeCrypto },
+      })
+      onNext(DefiStep.Confirm)
+      contextDispatch({ type: OsmosisWithdrawActionType.SET_LOADING, payload: false })
+    },
+    [
+      state,
+      contextDispatch,
+      lpAsset?.precision,
+      osmosisOpportunity,
+      underlyingAsset0,
+      underlyingAsset1,
+      calculateTokenOutMins,
+      getWithdrawFeeEstimate,
+      onNext,
+    ],
+  )
+
+  const handleCancel = browserHistory.goBack
+
+  const handlePercentClick = (percent: number) => {
+    if (!opportunityBalances) return
+    const cryptoAmountBaseUnit = bnOrZero(lpAssetBalance).multipliedBy(percent).toString()
+    const cryptoAmountPrecision = bnOrZero(cryptoAmountBaseUnit)
+      .dividedBy(bnOrZero(bn(10).pow(lpAsset?.precision ?? '0')))
+      .toString()
+    const fiatAmount = bnOrZero(opportunityBalances?.fiatBalance).multipliedBy(percent).toString()
+
+    setValue(Field.FiatAmount, fiatAmount, { shouldValidate: true })
+    setValue(Field.CryptoAmount, cryptoAmountPrecision, { shouldValidate: true })
+    handleInputChange(cryptoAmountBaseUnit, false)
+  }
+
+  useEffect(() => {
+    // Calculate underlying token balances
+    if (!lpAsset) return
+    ;(async () => {
+      const balances = await calculateBalances(lpAsset, lpAssetBalance)
+      if (!balances) return
+      setOpportunityBalances(balances)
+    })()
+
+    // Fetch pool data and calculate "price" of LP asset
+    const getPoolAssetMarketData = async () => {
+      /* No market exists for Osmosis pool assets, but we can calculate the 'price' of each pool token
+    by dividing the pool TVL by the total number of pool tokens. */
+      if (!(state && state.opportunity)) return undefined
+
+      const { assetReference: poolAssetReference } = fromAssetId(state.opportunity.assetId)
+      const id = getPoolIdFromAssetReference(poolAssetReference)
+      if (!id) return undefined
+
+      if (!poolData) setPoolData(await getPool(id))
+      if (!(poolData && poolData.total_shares)) return undefined
+
+      setPoolAssetMarketData({
```

Comment:
Fine for now so we get this in in its current state, but we should do this at `marketData` level so this gets properly inserted, we don't have to skip Osmo LP asset and we properly get marketData for them